### DISCUSSION TITLE
Page is undefined

### DISCUSSION THREAD

**Discussion by enri90** [vikejs/vike#814] [Discussion] [answered]
i've been having problems for days to try a production version of vitejs ssr urql express

I was using ts-node-dev in commonjs
now ts-node in esm

Warning: React.jsx: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: undefined. You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.

_default.page.server.tsx amd _default.page.client.tsx
Page is undefined

```typescript
  const pageContent = (
      <PageShell pageContext={pageContext}>
        <Provider value={client}>
          <Page {...pageProps} />
        </Provider>
      </PageShell>
    )
``` 

```typescript
"type": "module",
"scripts": {
    "build": "vite build",
    "dev": "ts-node ./server/index.ts",
    "server": "vite build && cross-env NODE_ENV=production ts-node ./server/index.ts",
``` 

```typescript
 "ts-node": {
    "transpileOnly": true,
    "esm": true,
    "compilerOptions": {
      "moduleResolution": "Node16"
    }
  },
```

**Comment by brillout** [comment]
Make sure to always pass components to Reacts, e.g. `assert(pageContext.Page)`.

**Comment by enri90** [comment, marked as answer]
migration Commonjs on Esm work

```typescript
const sirv = require('sirv')
``` 

```typescript
const sirv = (await import('sirv')).default
``` 

thx
